tree/30_generic_methods/service_started.cf

```diff
@@ -28,12 +28,10 @@ bundle agent service_started(service_name)
 {
   vars:
     "canonified_service_name"      string => canonify("${service_name}");
-
     "old_class_prefix"             string => "service_started_${canonified_service_name}";
-    "promisers"          slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
-    "class_prefix"      string => canonify(join("_", "promisers"));
     "args"                          slist => { "${service_name}" };
-
+    "report_param"                 string => join("_", args);
+    "class_prefix"                 string => canonify("${old_class_prefix}_${report_param}");
```

Comment:
There's only one args, so canonify("service_started_${report_param}"); or directly ${old_class_prefix} would do the trick